include/path_tracking_pid/path_tracking_pid_local_planner.hpp

```diff
@@ -90,10 +90,9 @@ class TrackingPidLocalPlanner : public mbf_costmap_core::CostmapController,
 private:
   /**
    * @brief Calculates the velocity command based on the current robot pose given by pose.
-   * @param cmd_vel Output the velocity command
-   * @return true if succeded.
+   * @return Velocity command on success, empty optional otherwise.
    */
-  bool computeVelocityCommands(geometry_msgs::Twist & cmd_vel);
```

Comment:
This whole function only existed for the sake of the `nav_core` implementation.

Now that we don't have that legacy anymore. I think there are much better improvements to be made.

No we're calling `computeVelocityCommands` from the `computeVelocityCommands` function. It doesn't add any semantics. The 'main' `computeVelocityCommands` could be broken into functions with better descriptions.

Reply:
I agree there are more and better improvements to be made. Nonetheless, this is also an improvement.  It may be small, but it's an improvement - at least from my perspective.

The investment has already been made and this won't block any other improvements from being made.